code/game/mecha/equipment/tools/other_tools.dm

```diff
@@ -0,0 +1,494 @@
+	selectable = 0
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/Destroy()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/detach()
+	processing_objects.Remove(src)
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/proc/get_charge()
+	if(equip_ready) //disabled
+		return
+	var/area/A = get_area(chassis)
+	var/pow_chan = get_power_channel(A)
+	if(pow_chan)
+		return 1000 //making magic
+
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/proc/get_power_channel(var/area/A)
+	var/pow_chan
+	if(A)
+		for(var/c in use_channels)
+			if(A.powered(c))
+				pow_chan = c
+				break
+	return pow_chan
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/Topic(href, href_list)
+	..()
+	if(href_list["toggle_relay"])
+		if(equip_ready) //inactive
+			processing_objects.Add(src)
+			set_ready_state(0)
+			log_message("Activated.")
+		else
+			processing_objects.Remove(src)
+			set_ready_state(1)
+			log_message("Deactivated.")
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/get_equip_info()
+	if(!chassis) return
+	return "<span style=\"color:[equip_ready?"#0f0":"#f00"];\">*</span>&nbsp; [name] - <a href='?src=\ref[src];toggle_relay=1'>[equip_ready?"A":"Dea"]ctivate</a>"
+
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/process()
+	if(!chassis || chassis.internal_damage & MECHA_INT_SHORT_CIRCUIT)
+		processing_objects.Remove(src)
+		set_ready_state(1)
+		return
+	var/cur_charge = chassis.get_charge()
+	if(isnull(cur_charge) || !chassis.cell)
+		processing_objects.Remove(src)
+		set_ready_state(1)
+		occupant_message("No powercell detected.")
+		return
+	if(cur_charge < chassis.cell.maxcharge)
+		var/area/A = get_area(chassis)
+		if(A)
+			var/pow_chan
+			for(var/c in list(EQUIP,ENVIRON,LIGHT))
+				if(A.powered(c))
+					pow_chan = c
+					break
+			if(pow_chan)
+				var/delta = min(20, chassis.cell.maxcharge-cur_charge)
+				chassis.give_power(delta)
+				A.use_power(delta*coeff, pow_chan)
+
+/////////////////////////////////////////// GENERATOR /////////////////////////////////////////////
+
+/obj/item/mecha_parts/mecha_equipment/generator
+	name = "exosuit plasma converter"
+	desc = "An exosuit module that generates power using solid plasma as fuel. Pollutes the environment."
+	icon_state = "tesla"
+	origin_tech = "plasmatech=2;powerstorage=2;engineering=2"
+	range = MELEE
+	var/coeff = 100
+	var/fuel_type = MAT_PLASMA
+	var/max_fuel = 150000
+	var/fuel_name = "plasma" // Our fuel name as a string
+	var/fuel_amount = 0
+	var/fuel_per_cycle_idle = 10
+	var/fuel_per_cycle_active = 100
+	var/power_per_cycle = 30
+
+/obj/item/mecha_parts/mecha_equipment/generator/New()
+	..()
+	generator_init()
+
+/obj/item/mecha_parts/mecha_equipment/generator/Destroy()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/item/mecha_parts/mecha_equipment/generator/proc/generator_init()
+	fuel_amount = 0
+
```

Comment:
Why is this necessary, if all it does is set `fuel_amount` to 0, which is its default value?